pymc_marketing/metrics.py

```diff
@@ -0,0 +1,93 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Evaluation Metrics."""
+
+import numpy as np
+import numpy.typing as npt
+
+
+def per_observation_crps(y_true: npt.NDArray, y_pred: npt.NDArray) -> npt.NDArray:
+    """Compute the continuous ranked probability score (CRPS) for each observation.
+
+    The CRPS — Continuous Ranked Probability Score — is a score function that compares a
+    single ground truth value to a Cumulative Distribution Function.
+
+    Parameters
+    ----------
+    y_true : array-like
+        The ground truth values.
+    y_pred : array-like
+        The predicted values. It is expected that y_pred has one extra sample
+        dimension on the left.
```

Comment:
Can you add an example of what these dims look like in the docstring. For instance, (date, ) and (sample, date) or (model, date) and (sample, model, date) 

Also that the leftmost dim is likely the samples? Is that the case for our models?

Reply:
I added an example in https://github.com/pymc-labs/pymc-marketing/pull/971/commits/d70e77ec88d5b85306fc7d9d419644c8cad55b1a . Hopefully it clarifies (It was confusing for me)